analytics/src/analytics/integrations/etldb/main.py

```diff
@@ -0,0 +1,134 @@
+    parent_path = Path(__file__).resolve().parent
+    sql_path = f"{parent_path}/create_etl_db.sql"
+
+    # read sql file
+    with open(sql_path) as f:
+        sql = f.read()
+
+    # execute sql
+    db = EtlDb()
+    cursor = db.connection()
+    cursor.execute(
+        text(sql),
+    )
+    db.commit(cursor)
+
+
+def sync_db(dataset: EtlDataset, effective: str) -> None:
+    """Write github data to etl database."""
+    # initialize a map of github id to db row id
+    ghid_map: dict[EtlEntityType, dict[str, int]] = {
+        EtlEntityType.DELIVERABLE: {},
+        EtlEntityType.EPIC: {},
+        EtlEntityType.SPRINT: {},
+        EtlEntityType.QUAD: {},
+    }
+
+    # initialize db connection
+    db = EtlDb(effective)
+
+    # sync quad data to db resulting in row id for each quad
+    ghid_map[EtlEntityType.QUAD] = sync_quads(db, dataset)
+    print(f"quad row(s) processed: {len(ghid_map[EtlEntityType.QUAD])}")
+
+    # sync deliverable data to db resulting in row id for each deliverable
+    ghid_map[EtlEntityType.DELIVERABLE] = sync_deliverables(
+        db,
+        dataset,
+        ghid_map,
+    )
+    print(f"deliverable row(s) processed: {len(ghid_map[EtlEntityType.DELIVERABLE])}")
+
+    # sync sprint data to db resulting in row id for each sprint
+    ghid_map[EtlEntityType.SPRINT] = sync_sprints(db, dataset, ghid_map)
+    print(f"sprint row(s) processed: {len(ghid_map[EtlEntityType.SPRINT])}")
+
+    # sync epic data to db resulting in row id for each epic
+    ghid_map[EtlEntityType.EPIC] = sync_epics(db, dataset, ghid_map)
+    print(f"epic row(s) processed: {len(ghid_map[EtlEntityType.EPIC])}")
+
+    # sync issue data to db resulting in row id for each issue
+    issue_map = sync_issues(db, dataset, ghid_map)
+    print(f"issue row(s) processed: {len(issue_map)}")
+
+
+def sync_deliverables(db: EtlDb, dataset: EtlDataset, ghid_map: dict) -> dict:
+    """Insert or update (if necessary) a row for each deliverable and return a map of row ids."""
+    result = {}
+    model = EtlDeliverableModel(db)
+    for ghid in dataset.get_deliverable_ghids():
+        deliverable_df = dataset.get_deliverable(ghid)
+        result[ghid], _ = model.sync_deliverable(deliverable_df, ghid_map)
+        if VERBOSE:
+            print(f"DELIVERABLE '{ghid}' row_id = {result[ghid]}")
+    return result
+
+
+def sync_epics(db: EtlDb, dataset: EtlDataset, ghid_map: dict) -> dict:
+    """Insert or update (if necessary) a row for each epic and return a map of row ids."""
+    result = {}
+    model = EtlEpicModel(db)
+    for ghid in dataset.get_epic_ghids():
+        epic_df = dataset.get_epic(ghid)
+        result[ghid], _ = model.sync_epic(epic_df, ghid_map)
+        if VERBOSE:
+            print(f"EPIC '{ghid}' row_id = {result[ghid]}")
+    return result
+
+
+def sync_issues(db: EtlDb, dataset: EtlDataset, ghid_map: dict) -> dict:
+    """Insert or update (if necessary) a row for each issue and return a map of row ids."""
```

Comment:
This most recent set of changes fixed the timeout beautifully!
<img width="1076" alt="Screenshot 2024-11-04 at 10 11 35 AM" src="https://github.com/user-attachments/assets/ac8add39-5775-4030-8036-bc7586e9a9fd">

Reply:
However, when I was testing the history table, I noticed something kind of strange -- after changing the point value of a given ticket, I saw the point value change in the `gh_issue_history` table but there was still only one row for the issue whose point value I changed:

<img width="1048" alt="Screenshot 2024-11-04 at 10 14 29 AM" src="https://github.com/user-attachments/assets/6fb8bbf3-63a5-4661-86d4-76ef18db51f8">

Am I misunderstanding how that table is supposed to work?

Reply:
Nevermind see the following comment!